ssl/ssl_ciph.c

```diff
@@ -2218,11 +2228,17 @@ int ssl_cipher_get_overhead(const SSL_CIPHER *c, size_t *mac_overhead,
 
 int ssl_cert_is_disabled(SSL_CTX *ctx, size_t idx)
 {
-    const SSL_CERT_LOOKUP *cl = ssl_cert_lookup_by_idx(idx);
+    /* A new provider-loaded key type is always enabled */
+    if (idx >= SSL_PKEY_NUM) {
+        return 0;
+    }
+    else {
```

Comment:
I'd ditch the else here.  The return in the if kind of makes it redundant.

Reply:
so changed in https://github.com/openssl/openssl/pull/19312/commits/e5582173f0db49a28edf0907db053ea66e10d3dc